src/multichainhelpers.ts

```diff
@@ -1,46 +1,42 @@
-import { ChainId, PublicKeyBundle } from "@iov/base-types";
-import { BcpAccount, SendTx, TokenTicker, TransactionKind } from "@iov/bcp-types";
-import { Address, MultiChainSigner } from "@iov/core";
-import { PublicIdentity } from "@iov/keycontrol";
+import {
+  Amount,
+  BcpAccount,
+  PublicIdentity,
+  PublicKeyBundle,
+  SendTransaction,
+  TokenTicker,
+} from "@iov/bcp-types";
+import { Address, MultiChainSigner, UserProfile } from "@iov/core";
 
 import { needsRefill, refillAmount } from "./cashflow";
 import { debugAccount, logAccountsState, logSendJob } from "./debugging";
 
-export function identitiesOfFirstChain(signer: MultiChainSigner): ReadonlyArray<PublicIdentity> {
-  const wallet = signer.profile.wallets.value[0];
-  return signer.profile.getIdentities(wallet.id);
+export function identitiesOfFirstChain(profile: UserProfile): ReadonlyArray<PublicIdentity> {
+  const wallet = profile.wallets.value[0];
+  return profile.getIdentities(wallet.id);
 }
 
 export function identityToAddress(signer: MultiChainSigner, identity: PublicIdentity): Address {
-  const chainId = signer.chainIds()[0];
-  return signer.keyToAddress(chainId, identity.pubkey);
+  return signer.identityToAddress(identity);
 }
 
-export async function accountsOfFirstChain(signer: MultiChainSigner): Promise<ReadonlyArray<BcpAccount>> {
-  const addresses = identitiesOfFirstChain(signer).map(identity => identityToAddress(signer, identity));
+export async function accountsOfFirstChain(
+  profile: UserProfile,
+  signer: MultiChainSigner,
+): Promise<ReadonlyArray<BcpAccount>> {
+  const addresses = identitiesOfFirstChain(profile).map(identity => signer.identityToAddress(identity));
   const chainId = signer.chainIds()[0];
 
   // tslint:disable-next-line:readonly-array
   const out: BcpAccount[] = [];
   for (const address of addresses) {
     const responsesPromise = signer.connection(chainId).getAccount({ address: address });
```

Comment:
I don't remember why I store the unresolved promise here. This makes no sense.

`const responses = await signer.connection(chainId).getAccount({ address: address });`